File: tests/fixtures/openai-realtime/openai-realtime-no-log-raw-message-payloads-fixed/01-debug-with-derived-field.ts
import WebSocket from 'ws';

function setup(logger: any) {
  const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
  const callerSocket = new WebSocket(url, { headers: {} });

  callerSocket.on('message', (raw) => {
    const message = JSON.parse(raw);
    logger.debug({ type: message.type }, 'Caller message from OpenAI');
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-no-log-raw-message-payloads-fixed/01_debug_with_derived_field.py
import json
import logging

import websockets

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"
logger = logging.getLogger(__name__)


async def stream():
    async with websockets.connect(WS_URL) as ws:
        async for message in ws:
            event = json.loads(message)
            logger.info("received event type=%s", event.get("type"))
